src/dmd/root/file.d

```diff
@@ -281,63 +652,50 @@ nothrow:
-            // Doesn't exist, not a regular file, different size
-            if (nameStr.extendedPathThen!(p => GetFileAttributesExW(p.ptr, GET_FILEEX_INFO_LEVELS.GetFileExInfoStandard, &fad)) == 0 ||
-                fad.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY ||
-                ((cast(ulong) fad.nFileSizeHigh << 32) | fad.nFileSizeLow) != data.length)
-            {
-                return write(namez, data);               // write new file
-            }
-        }
-        else version (Posix)
-        {
-            import core.sys.posix.sys.stat;
+        if (data.length != File.size(namez))
+            return write(namez, data);               // write new file
 
-            stat_t statbuf = void;
-            if (stat(namez, &statbuf) != 0 ||            // doesn't exist
-                (statbuf.st_mode & S_IFMT) != S_IFREG || // not a regular file
-                statbuf.st_size != data.length)          // different size
-            {
-                if (log) printf("not exist or diff size %d %d %d\n",
-                    stat(namez, &statbuf) != 0,
-                    (statbuf.st_mode & S_IFMT) != S_IFREG,
-                     statbuf.st_size != data.length);
-                return write(namez, data);               // write new file
-            }
-        }
-        else
-            static assert(0);
         if (log) printf("same size\n");
 
         /* The file already exists, and is the same size.
          * Read it in, and compare for equality.
-         * For larger files, this could be faster by comparing the file
-         * block by block and quitting on first difference.
          */
-        ReadResult r = read(namez);
-        if (!r.success ||
-            r.buffer.data[] != data[])
+        //if (FileMapping!(const ubyte)(namez)[] != data[])
             return write(namez, data); // contents not same, so write new file
-        if (log) printf("same contents\n");
+        //if (log) printf("same contents\n");
 
         /* Contents are identical, so set timestamp of existing file to current time
          */
+        //return touch(namez);
+    }
+
+    ///ditto
+    extern(D) static bool update(const(char)[] name, const void[] data)
+    {
+        return name.toCStringThen!(fname => update(fname.ptr, data));
+    }
+
+    /// ditto
+    extern (C++) static bool update(const(char)* name, const(void)* data, size_t size)
+    {
+        return update(name, data[0 .. size]);
+    }
+
+    /// Touch a file to current date
+    static bool touch(const char* namez)
+    {
         version (Windows)
         {
             FILETIME ft = void;
             SYSTEMTIME st = void;
             GetSystemTime(&st);
             SystemTimeToFileTime(&st, &ft);
 
+            import core.stdc.string : strlen;
+
             // get handle to file
-            HANDLE h = nameStr.extendedPathThen!(p => CreateFile(p.ptr,
+            HANDLE h = namez[0 .. namez.strlen].extendedPathThen!(p => CreateFile(p.ptr,
```

Comment:
don't really like omitting the parens on function calls

Reply:
it's your feature :)